src/plugins/workspace/public/components/workspace_form/workspace_cancel_modal.tsx

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React from 'react';
+import { i18n } from '@osd/i18n';
+import { EuiConfirmModal } from '@elastic/eui';
+import { ApplicationStart } from 'opensearch-dashboards/public';
+import { WORKSPACE_LIST_APP_ID } from '../../../common/constants';
+
+interface WorkspaceCancelModalProps {
+  visible: boolean;
+  application: ApplicationStart;
+  closeCancelModal: () => void;
+}
+
+export const WorkspaceCancelModal = ({
+  application,
+  visible,
+  closeCancelModal,
+}: WorkspaceCancelModalProps) => {
+  if (!visible) {
```

Comment:
I think we should move `visible` out and use it like this:
```typescript
      {isCancelModalVisible && <WorkspaceCancelModal
        application={application}
        closeCancelModal={closeCancelModal}
      />}
```

Reply:
`visible` property can be removed.